controllers/argocd/argocd_controller_test.go

```diff
@@ -102,6 +103,82 @@ func TestReconcileArgoCD_Reconcile(t *testing.T) {
 	}
 }
 
+func TestReconcileArgoCD_Reconcile_withRemoveManagedByLabelOnArgocdDeletionNotSet(t *testing.T) {
+	restoreEnv(t)
+	logf.SetLogger(ZapLogger(true))
+	a := makeTestArgoCD(deletedAt(time.Now()), addFinalizer(common.ArgoCDDeletionFinalizer))
+	r := makeTestReconciler(t, a)
+
+	nsArgocd := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+		Name: a.Namespace,
+	}}
+	err := r.Client.Create(context.TODO(), nsArgocd)
+	assert.NoError(t, err)
+
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+		Name: "newNamespaceTest",
+		Labels: map[string]string{
+			common.ArgoCDManagedByLabel: a.Namespace,
+		}},
+	}
+	err = r.Client.Create(context.TODO(), ns)
+	assert.NoError(t, err)
+
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      a.Name,
+			Namespace: a.Namespace,
+		},
+	}
+
+	_, err = r.Reconcile(context.TODO(), req)
+	assert.NoError(t, err)
+
+	// Check if the managed-by label still exists in the new namespace
+	assert.NoError(t, r.Client.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, ns))
+	assert.Equal(t, ns.Labels[common.ArgoCDManagedByLabel], a.Namespace)
+}
+
+func TestReconcileArgoCD_Reconcile_withRemoveManagedByLabelOnArgocdDeletionTrue(t *testing.T) {
+	restoreEnv(t)
+	logf.SetLogger(ZapLogger(true))
+	a := makeTestArgoCD(deletedAt(time.Now()), addFinalizer(common.ArgoCDDeletionFinalizer))
+	r := makeTestReconciler(t, a)
+
+	os.Setenv("REMOVE_MANAGED_BY_LABEL_ON_ARGOCD_DELETION", "true")
+
+	nsArgocd := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+		Name: a.Namespace,
+	}}
+	err := r.Client.Create(context.TODO(), nsArgocd)
+	assert.NoError(t, err)
+
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+		Name: "newNamespaceTest",
+		Labels: map[string]string{
+			common.ArgoCDManagedByLabel: a.Namespace,
+		}},
+	}
+	err = r.Client.Create(context.TODO(), ns)
+	assert.NoError(t, err)
+	
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      a.Name,
+			Namespace: a.Namespace,
+		},
+	}
+
+	_, err = r.Reconcile(context.TODO(), req)
+	assert.NoError(t, err)
+
+	// Check if the managed-by label gets removed from the new namespace
+	assert.NoError(t, r.Client.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, ns))
+	if _, ok := ns.Labels[common.ArgoCDManagedByLabel]; ok {
+		t.Errorf("Expected the label[%v] to be removed from the namespace[%v]", common.ArgoCDManagedByLabel, ns.Name)
+	}
+}
+
```

Comment:
Can we use a table format for these tests instead 
that would help avoid redundancy of testing code 
please refer to this example https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/status_test.go#L55